### DISCUSSION TITLE
Help getting TIC-80 to work on RetroPie 4.7.1

### DISCUSSION THREAD

**Discussion by MBRetrocade** [nesbox/TIC-80#1321] [Q&A]
Hello everyone, so I am new to this and am in need of some help.

I wrote my first script module to download/install/make TIC-80 on RetroPie 4.7.1

It unfortunately doesn't work. I managed to get the script to 'play' through and 'install'. But TIC-80 won't boot.

I did grab tic80_libretro.so off a Recalbox build. 

Has anyone else managed to (successfully) install TIC-80 on a Raspberry Pi 4 running RetroPie 4.7.1?
Any help would be appreciated! I was wanting to put this together as a birthday gift. 

**Here are the contents of: lr-tic80.sh**

#!/usr/bin/env bash

# This file is part of The RetroPie Project
#
# The RetroPie Project is the legal property of its developers, whose names are
# too numerous to list here. Please refer to the COPYRIGHT.md file distributed with this source.
#
# See the LICENSE.md file at the top-level directory of this distribution and
# at https://raw.githubusercontent.com/RetroPie/RetroPie-Setup/master/LICENSE.md
#

rp_module_id="lr-tic80"
rp_module_desc="TIC-80 is a fantasy computer for making, playing and sharing tiny games"
rp_module_help="Copy your TIC-80 games to $romdir/tic80"
rp_module_licence="GPL2 https://raw.githubusercontent.com/libretro/TIC-80/master/LICENSE"
rp_module_section="exp"
rp_module_flags="!armv6"

function sources_lr-tic80() {
    gitPullOrClone "$md_build" https://github.com/libretro/TIC-80.git	
}

function build_lr-tic80() {
    local params=()
    isPlatform "arm" && params+=(platform=armvneon)
    cd lr-tic80/src/libretro
    make clean
    make "${params[@]}"
    md_ret_require="$md_build/lr-tic80/src/libretro/tic80_libretro.so"
}

function install_lr-tic80() {
    md_ret_files=(
        'lr-tic80/src/libretro/tic80_libretro.so'
        'LICENSE'
        'README.md'
    )
}

function configure_lr-tic80() {
    mkRomDir "tic80"
    ensureSystemretroconfig "tic80"

    addEmulator 1 "$md_id" "tic80" "$md_inst/tic80_libretro.so"
    addSystem "tic80" "TIC-80" ".tic .TIC"
}

**Here are the contents of: es_systems.cfg**
  <system>
    <name>tic80</name>
    <fullname>TIC-80</fullname>
    <path>/home/pi/RetroPie/roms/tic80</path>
    <extension>.tic .TIC</extension>
    <command>/opt/retropie/supplementary/runcommand/runcommand.sh 0 _SYS_ tic80 %ROM%</command>
    <platform>tic80</platform>
    <theme>tic80</theme>
  </system>
  
  **Here are the contents of: emulators.cfg**
  lr-tic80 = "/opt/retropie/emulators/retroarch/bin/retroarch -L /opt/retropie/libretrocores/lr-tic80/tic80_libretro.so --config /opt/retropie/configs/tic80/retroarch.cfg %ROM%"
default = "lr-tic80"

**Here are the contents of: retroarch.cfg**
# Settings made here will only override settings in the global retroarch.cfg if placed above the #include line

input_remapping_directory = "/opt/retropie/configs/tic80/"

#include "/opt/retropie/configs/all/retroarch.cfg"

**Comment by MBRetrocade** [comment]
Hopefully you can understand what I pasted above. It got a little funky after pasting. Any help would be greatly appreciated! Thank you

**Comment by nesbox** [comment]
The only @RobLoach can help you :)

**Reply by MBRetrocade** [reply]
Thank you :) We would love to play this on our RetroPie build. I am looking forward to getting it done for his birthday. It would make a nice surprise :). 

Thank you for tagging @RobLoach I am looking forward to his insight

**Comment by SuperPupperDoggo** [comment]
is there a way to have it work on a Pi Zero as well?
